### ISSUE TITLE
JSX rules don't work with hand-written React `createElement` code

### ISSUE THREAD

**Issue by jaydenseric** [denoland/deno_lint#1433]
### Lint Name

Probably all the JSX rules, but I was trying [`jsx-key`](https://docs.deno.com/lint/rules/jsx-key/).

### Code Snippet

```json
{
  "lock": false,
  "compilerOptions": {
    "lib": [
      "dom",
      "dom.iterable",
      "dom.asynciterable",
      "deno.ns",
      "deno.unstable"
    ]
  },
  "lint": {
    "rules": {
      "include": ["jsx-key", "react-rules-of-hooks"]
    }
  }
}
```

```ts
import { createElement as h } from "react";

const foo = [1, 2, 3].map(() => h("div"));
```

### Expected Result

A lint error that a `key` is needed in the mapped divs.

### Actual Result

No lint error about JSX keys.

### Additional Info

In projects and published packages I write type safe vanilla JS modules via TypeScript JSDoc, even for React projects, and it makes no sense that you are forced to introduce a transpilation build step just to use React JSX rules in Deno lint.

Some examples:

- Published in Ruck: https://github.com/jaydenseric/ruck/blob/8e78790f253f0b02f1473b9d7425c26d14ad4990/Html.mjs
- A deployed project: https://github.com/jaydenseric/device-agnostic-ui-website/blob/cb027cfd73883a77189d599980753276462ab636/public/components/App.mjs

### Version

deno 2.2.6 (stable, release, aarch64-apple-darwin)
v8 13.5.212.4-rusty
typescript 5.7.3
